api/app/Models/PoolCandidate.php

```diff
@@ -834,4 +839,112 @@ public function setApplicationSnapshot()
         $this->profile_snapshot = $profile;
         $this->save();
     }
+
+    public function computeDecisionsPerAssessmentStep()
+    {
+        $decisions = [];
+        $this->load([
+            'pool.assessmentSteps',
+            'assessmentResults',
+        ]);
+
+        foreach ($this->pool->assessmentSteps as $step) {
+            $stepId = $step->id;
+            $hasFailure = false;
+            $hasOnHold = false;
+            $hasToAssess = false;
+
+            $stepResults = $this->assessmentResults()->where('assessment_step_id', $stepId);
+            if (! $stepResults->count()) {
+                $decisions[$stepId] = null;
+
+                continue;
+            }
+
+            // Check assessed essential skills on this step
+            $essentialSkillAssessments = $stepResults->whereHas('assessmentStep', function ($query) {
+                $query->whereHas('poolSkills', function ($query) {
```

Comment:
Make sure to preload `assessmentResults.assessmentStep.poolSkills`

Reply:
Good catch! Added in [preload pool skills for decision computation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10642/commits/4b521d5b631871b3e9e769578158241230ca1dea)